### ISSUE TITLE
warn about 'grep pattern' without specifying filename

### ISSUE THREAD

**Issue by bittorf** [koalaman/shellcheck#567]
like https://github.com/koalaman/shellcheck/issues/452 we should
warn about an forever hanging grep, e.g.

```
grep mypattern
```
or
```
grep -switches mypattern
```
with a forgotten filename.

**Comment by deryni** [comment]
GNU grep as of some version 2.11 ([changelog entry](http://git.savannah.gnu.org/cgit/grep.git/tree/NEWS#n433)) assumes that a missing filename means the current directory when run with the `-r, --recursive` flag.

**Comment by bittorf** [comment]
@deryni interesting - thanks, the feature is from march 2012 and it is not POSIX 8-) so still a valid case and a warning should be OK. for interactive scripting it's OK to leave the filename blank, but for real you should do a
```
grep -r PATTERN ./
```

**Comment by deryni** [comment]
I agree completely. I just wanted to point out that it is technically valid (for GNU grep).
